filebeat/fileset/pipelines.go

```diff
@@ -48,6 +49,19 @@ type MultiplePipelineUnsupportedError struct {
 	minESVersionRequired common.Version
 }
 
+// Processor defines a single processors minimum version requirements.
+type Processor struct {
```

Comment:
I don't see any reason to export this so make it `processor` or maybe even called it `processorCompatibility`.